Source/DafnyCore/Options/DafnyCommands.cs

```diff
@@ -90,6 +90,7 @@ static DafnyCommands() {
     Function.FunctionSyntaxOption,
     CommonOptionBag.QuantifierSyntax,
     CommonOptionBag.UnicodeCharacters,
+    CommonOptionBag.RawPointers,
```

Comment:
Two points there.
That approach already caused random exceptions because uninitialized static code did not run in the right order, like this one:
https://github.com/dafny-lang/dafny/pull/5685
So unless you want me to go ahead and apply the same fix that I did for the 10+ places where options were wrongly registered, I'd prefer to keep it here.

Second, raw pointers will mean that deallocation will be possible, although it's going to be another CLI option. Still, it might impact verification down the road.

Reply:
I've updated #5685 so I'm more confident with the fix now that I removed the possibility of the static initialization to be concurrent. I hope it works, if so, I'll go back to this option and initialize it the same way, thanks.

Reply:
> That approach already caused random exceptions because uninitialized static code did not run in the right order, like this one:
https://github.com/dafny-lang/dafny/pull/5685
So unless you want me to go ahead and apply the same fix that I did for the 10+ places where options were wrongly registered, I'd prefer to keep it here.

Fair enough, and I don't care too much about what file the option is declared in. I care more about only attaching it to `dafny translate rs` specifically since the name makes no sense for most of the other languages. I know it's hidden but I also know some of our customers at least run `dafny --help-internal` and would be confused by `--raw-pointers` being an option in `dafny translate java --help-internal`

> Second, raw pointers will mean that deallocation will be possible, although it's going to be another CLI option. Still, it might impact verification down the road.

If you wanted this to be a more general option that just means you can free objects, that's fine, but it needs a different name in that case. But it seems like this option enables a different translation strategy in Rust and `--raw-pointers` is probably better in this case I think.

Reply:
> Fair enough, and I don't care too much about what file the option is declared in.

The option should always be registered in the same file as option is declared in, so the registration is guaranteed to happen in time if the option is used.

I've added a comment to https://github.com/dafny-lang/dafny/pull/5685 because I think it's making an unnecessary change.

Reply:
All right, now I felt confident enough to move the option. I've also set up compatibility for library as "CheckOptionMatches" because it's not possible to combine raw pointers with objects as of today.

Reply:
I want this option also to be available in "dafny build", "dafny run" and "dafny test".
However, it seems I did not make it right, I keep getting a 
```
** Error: Command-line argument '--raw-pointers' is neither a recognized option nor a filename with a supported extension (.dfy, .rs).
```

This is despite attempts like in RustBackend

```
  public static readonly Option<bool> RustRawPointers = new("--raw-pointers", () => false,
    @"
false - All class instances are reference-counted
true - All class instances are raw pointers (future versions of Dafny will include deallocation)".TrimStart()) {
    IsHidden = true
  };

   public override IEnumerable<Option> SupportedOptions => new List<Option> { RustRawPointers };

  static RustBackend() {
    OptionRegistry.RegisterGlobalOption(RustRawPointers, OptionCompatibility.CheckOptionMatches);
  }
```

Variants I tested at the end include:
```
    OptionRegistry.RegisterOption(RustRawPointers, OptionScope.Translation);
```
and
```
    OptionRegistry.RegisterOption(RustRawPointers, OptionScope.Cli);
```
I don't know what I'm doing wrong. Python uses the second case just fine.

Reply:
The second argument to `OptionRegistry.RegisterOption` is about how to record the option in `.doo` or `.dtr` files, it doesn't affect how the option is attached to the actual CLI.

But you're right, `IExecutableBackend.SupportedOptions` is currently only hooked up to `dafny translate <id>` and not other commands like `dafny build -t:<id>`. That should be fixed so that existing options like `--go-module-name` are available to those commands too. Cut a separate issue for that, which hopefully we can address quickly before merging this PR: https://github.com/dafny-lang/dafny/issues/5707